stake-pool/program/src/instruction.rs

```diff
@@ -1423,6 +1427,73 @@ pub fn update_validator_list_balance(
     }
 }
 
+/// Creates an `UpdateValidatorListBalance` instruction (update validator stake
+/// account balances) to update `validator_list[start_index..start_index +
+/// len]`.
+///
+/// Returns `None` if:
+/// - `start_index..start_index + len` is out of bounds for
+///   `validator_list.validators`
+pub fn update_validator_list_balance_chunk(
```

Comment:
Just curious if you think `Option` or `Result` is better here. I know you mentioned both.

To me, it feels a bit like this should be a `Result`, since if the length is out of bounds, it's attempting to create an invalid instruction, which should be fallible. What do you think? Maybe you have some use cases in mind I can't think of.

Reply:
I just chose `Option` bec thats what `slice.get()` returns and i'm lazy lol. Also because im not sure what the error type should be if it was a `Result`

Reply:
How about `ProgramError::InvalidInstructionData`?

Reply:
Done. Yall could've just decided on this from the start instead of telling me to "add the Option or Result handling as you see fit" lol